junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -186,8 +186,19 @@ fun <R> assertDoesNotThrow(message: String, executable: () -> R): R =
  * @param R the result type of the [executable]
  */
 @API(status = EXPERIMENTAL, since = "5.5")
-fun <R> assertDoesNotThrow(message: () -> String, executable: () -> R): R =
-    Assertions.assertDoesNotThrow(ThrowingSupplier(executable), Supplier(message))
+inline fun <R> assertDoesNotThrow(noinline message: () -> String, executable: () -> R): R =
+    Assertions.assertDoesNotThrow(
+        getThrowingSupplier(executable),
+        Supplier(message)
+    )
+
+@PublishedApi
+internal inline fun <R> getThrowingSupplier(executable: () -> R): ThrowingSupplier<R> = try {
```

Comment:
Not sure about this method name which is "effectively public" due to `@PublishedApi`. Maybe `evaluateAndWrap` or sth. similar would be clearer.

Reply:
Yeah that's a good call, agreed. 

https://github.com/junit-team/junit5/pull/2684/commits/742d7af3258469c30e6a997d5c372d436cf7a762